### ISSUE TITLE
Median function does not work

### ISSUE THREAD

**Issue by jchnxu** [AlaSQL/alasql#1491]
Median function does not work? 

This is the fiddle. 
http://jsfiddle.net/m187t5vk/
```
["Error: TypeError: alasql.fn.median is not a function
    at alasql.Query.eval <a class='gotoLine' href='#as selectfn'>as selectfn</a> (eval at yy.Select.compileSelect2 (https://cdn.jsdelivr.net/npm/alasql@0.4/dist/alasql.js:9730:9), &lt;anonymous&gt;:3:51)
    at doJoin (https://cdn.jsdelivr.net/npm/alasql@0.4/dist/alasql.js:7452:27)
    at doJoin (https://cdn.jsdelivr.net/npm/alasql@0.4/dist/alasql.js:7536:8)
    at queryfn3 (https://cdn.jsdelivr.net/npm/alasql@0.4/dist/alasql.js:6954:2)
    at queryfn2 (https://cdn.jsdelivr.net/npm/alasql@0.4/dist/alasql.js:6928:9)
    at source.datafn (https://cdn.jsdelivr.net/npm/alasql@0.4/dist/alasql.js:8363:20)
    at https://cdn.jsdelivr.net/npm/alasql@0.4/dist/alasql.js:6876:19
    at Array.forEach (&lt;anonymous&gt;)
    at queryfn (https://cdn.jsdelivr.net/npm/alasql@0.4/dist/alasql.js:6872:16)
    at statement (https://cdn.jsdelivr.net/npm/alasql@0.4/dist/alasql.js:8020:14)"]
```

This is what happened on my machine. 
```
SELECT median(`total_count`) FROM CSV('/myproject/a893312d-9d41-4491-966f-f6a237a7a2e6.csv') AS `out_csv`
undefined:3
var y;var r={'(total_count)':(y=alasql.fn["median"](p['out_csv']['total_count']),y===y?y:undefined)};return r
                                                   ^

TypeError: alasql.fn.median is not a function
    at alasql.Query.eval [as selectfn] (eval at yy.Select.compileSelect2 (/myproject/node_modules/alasql/dist/alasql.fs.js:9665:9), <anonymous>:3:52)
    at doJoin (/myproject/node_modules/alasql/dist/alasql.fs.js:7459:27)
    at doJoin (/myproject/node_modules/alasql/dist/alasql.fs.js:7545:8)
    at queryfn3 (/myproject/node_modules/alasql/dist/alasql.fs.js:6966:2)
    at queryfn2 (/myproject/node_modules/alasql/dist/alasql.fs.js:6940:9)
    at parseText (/myproject/node_modules/alasql/dist/alasql.fs.js:17574:10)
    at /myproject/node_modules/alasql/dist/alasql.fs.js:3496:7
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3)
Debugger listening on ws://0.0.0.0:9232/c85e54b8-16f7-4449-bdfd-05c556ca648c
```

**Comment by mathiasrw** [comment]
Is `median` part of the SQL definition?

**Comment by jchnxu** [comment]
BTW, on the server it will throw a process.uncaughtExeption, breaking the server. There's no way to catch it. I believe we should be able to catch those in the callback error

**Comment by mathiasrw** [comment]
Released as part of v2.1.5
